### ISSUE TITLE
Error compiling on RHEL6

### ISSUE THREAD

**Issue by CRCinAU** [skeeto/endlessh#10]
Error text:

```
# make
cc -ggdb3 -std=c99 -Wall -Wextra -Wno-missing-field-initializers -Os -o endlessh endlessh.c 
endlessh.c: In function ‘main’:
endlessh.c:105: warning: dereferencing pointer ‘s’ does break strict-aliasing rules
endlessh.c:104: note: initialized from here
endlessh.c:110: warning: dereferencing pointer ‘s’ does break strict-aliasing rules
endlessh.c:109: note: initialized from here
/tmp/ccNO67G7.o: In function `uepoch':
/root/endlessh/endlessh.c:33: undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
make: *** [endlessh] Error 1
```

In the Makefile, the following changes fixed the compile:
```
-LDFLAGS = -ggdb3
+LDFLAGS = -ggdb3 -lrt
```

**Comment by skeeto** [comment]
Strictly speaking, the warnings about strict aliasing are basically 
correct. However, this is just Berkeley sockets type punning crap, 
necessary when using that API. The system C compiler should be aware of 
this and ultimately shouldn't produce these warnings. Indeed, more 
recent versions of Linux, including RHEL 7, don't.

As you noticed, older versions of glibc used to put clock_gettime() in 
librt. These functions were moved into libc in glibc 2.17 (December 
2012), so recent Linux distributions do not require -ltr when linking. 
Other standard C libraries, such as musl and BSD, don't even have a 
librt, so using -ltr would be an error.

I don't want to change the Makefile to handle a quirk of a lone, 
decade-old Linux distribution — probably the only one that still 
supports such an old glibc — when it would break a bunch of other 
systems. I've added a note to the README about this (44219a5). Though 
the variable to modify is LDLIBS, not LDFLAGS because gcc (and Clang's 
gcc driver) are sensitive to argument order.

Fortunately you don't need to modify the Makefile at all, just use a 
command line argument:

    make LDLIBS=-lrt
